### PR TITLE
Sonar to read lcov report from jest

### PR DESCRIPTION
**Pull request by yehjxraymond**

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 module.exports = {
   setupFiles: ["<rootDir>/jest.setup.js"],
   testPathIgnorePatterns: ["<rootDir>/.next/", "<rootDir>/node_modules/"],
-  collectCoverage: false,
+  collectCoverage: true,
```
Comment (rjchow) [inline jest.config.js]: is it possible to parameterise this? its a bit annoying for it to print coverage when i just want to see test results

Comment (yehjxraymond) [reply jest.config.js]: Naw... deal with it

Comment (yehjxraymond) [reply jest.config.js]: Jk... you can turn it on with a flag, so it can stay disabled.

File: scripts/sonar-runner.sh [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
+if ! [ "${TRAVIS_PULL_REQUEST}" = "false" ]; then
+    sonar-scanner \
+        -Dsonar.projectKey=OpenCerts_opencerts-website \
+        -Dsonar.organization=opencerts \
+        -Dsonar.sources=. \
```
Comment (rjchow) [inline scripts/sonar-runner.sh]: should this be src/ ?

File: sonar-project.properties [hunk 3]
```diff
@@ -2,9 +2,10 @@ sonar.projectKey=OpenCerts_opencerts-website
 sonar.projectName=OpenCerts_opencerts-website
 sonar.projectVersion=0.0.1
 sonar.sources=src
+sonar.tests=src
+sonar.test.inclusions=src/**/*.spec.js,src/**/*.spec.jsx,src/**/*.test.js,src/**/*.test.jsx
```
Comment (rjchow) [inline sonar-project.properties]: does coverage calculation exclude tests? for mcf I had to specifically exclude it iirc

Comment (yehjxraymond) [reply sonar-project.properties]: https://sonarcloud.io/component_measures?id=OpenCerts_opencerts-website&metric=coverage&view=list

It ingest report from jest anyway, turns out jest is printing coverage in the scripts folder only. Seemed to be a jest config issue.

File: scripts/sonar-runner.sh (hunk 2 above)
Comment (yehjxraymond) [reply scripts/sonar-runner.sh]: There is scripts in `scripts`

**Comment by rjchow** [comment]
approved

**Pull request merged**